build_docs.py

```diff
@@ -1053,28 +1003,60 @@ def purge_path(www_root: Path, path: Path):
     run(["curl", "-XPURGE", f"https://docs.python.org/{{{','.join(to_purge)}}}"])
 
 
+def parse_versions_from_devguide():
+    releases = requests.get(
+        "https://raw.githubusercontent.com/"
+        "python/devguide/main/include/release-cycle.json",
+        timeout=30,
+    ).json()
+    return [Version.from_json(name, release) for name, release in releases.items()]
+
+
+def parse_languages_from_config():
+    """Read config.toml to discover languages to build."""
+    config = tomlkit.parse((HERE / "config.toml").read_text(encoding="UTF-8"))
+    languages = []
+    for name, section in config["languages"].items():
+        languages.append(
+            Language(
+                name,
+                section.get("iso639_tag", name),
+                section["name"],
+                section.get("in_prod", config["defaults"]["in_prod"]),
+                sphinxopts=section.get("sphinxopts", config["defaults"]["sphinxopts"]),
+                html_only=section.get("html_only", config["defaults"]["html_only"]),
```

Comment:
Maybe save the defaults to locals to avoid a lookup every time?

Reply:
Done, that's a bit more readable, thanks!